### DISCUSSION TITLE
H1 registers for inverter time.

### DISCUSSION THREAD

**Discussion by Ashthos** [nathanmarlor/foxess_modbus#293] [Q&A] [answered]
Hi, does anyone know the Modbus registers for the (date &) time of the inverter?

I'd like to be able to set the new time when the clocks change, as the inverter runs the charge periods based on its internal clock.

Thanks.

**Comment by canton7** [comment, marked as answer]
Yep, see here (up in the 41000 range): https://github.com/TonyM1958/HA-FoxESS-Modbus/wiki/Modbus-Map.xls

**Reply by canton7** [reply]
I'd be open to adding a service to set this. I don't really want to add it as a sensor, as it'll just be changing every poll, which will fill up your HA database with fairly useless values.

**Reply by Ashthos** [reply]
I agree. I only plan on setting the current time periodically, say, once per week, to prevent clock drift. I don't care what the setting is before I set it. There is no value in knowing what the clock drift is, as I can do nothing to prevent it - apart from setting the current time, which just really sets the current clock drift to zero.

**Comment by TonyM1958** [comment]
Is there evidence for clock drift on the inverter? Mine updated automatically around force charge times for BST, for example, with no intervention.

So far as I can tell, as long as you have a working connection on the data logger, the inverter time seems to maintain synchronisation. so there may be running some form of NTP running via the data logger... from a few API related things I think I've seen, Fox are pretty keen on maintaining a level of synchronisation between the inverter and cloud.

If you manually change the inverter time, do you know what this might do to anything that is running on the inverter and/or cloud?

**Reply by FozzieUK** [reply]
I think the internal clock itself is reasonably accurate (a few seconds a month) but if you have the datalogger fitted it will synchronise occasionally with the Fox Cloud and at times of DST change (I haven't tested to see what the NTP interval is), the most i've ever seen it out is by 2 seconds.

Time is important to the system (but not critical), mostly to do with accuracy of statistics which will be messed up if you make significant manual changes.  Time is sync'd across the system, the inverter broadcast time every 6 seconds to the BMS, the battery packs themselves don't care that much about time but they do get a regular heartbeat timer from the BMS.

**Reply by screenagerbe** [reply]
On a 5 month period, I would say the internal clock is about 8 min behind.
I disconnected my inverter from the internet the end of May. I reconnected it today in the hope DST would be applied, but alas ... I had to set it manually and that's when I saw the time was 52 minutes ahead when it should be 1:00 :)

So @canton7, any news on that time/date integration? 😇
I would love to be able to read the internal time of the inverter and set it via HA.

**Reply by canton7** [reply]
Opened #463
